src/autopas/utils/AutoPasMacros.h

```diff
@@ -0,0 +1,14 @@
+/**
+ * @file AutoPasMacros.h
+ * @author seckler
+ * @date 14.05.19
+ */
+
+#pragma once
+
+#ifdef __GNUC__
+/**
+ * @todo c++17: __attribute__((warn_unused_result)) should be replaced with [[nodiscard]]
+ */
+#define AUTOPAS_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
```

Comment:
clang version?

Reply:
clang apparently also defines __GNUC__, but added __clang__ anyway. 
also tested it. __attribute__((warn_unused_result)) works with clang as well.